google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.java

```diff
@@ -164,6 +190,8 @@ public ApiFuture<AppendRowsResponse> append(JSONArray jsonArr, long offset)
           } else {
             rowIndexToErrorMessage.put(i, exception.getMessage());
           }
+        } catch (InterruptedException ex) {
+          throw new RuntimeException(ex);
```

Comment:
AFAIU this is changing the behavior for downstream consumers who may handle exceptions differently, like Datastream, as now the exception is wrapped. please consult with @idobar on whether this affecting us.

Reply:
I added one immediate retry and then a second try after 30 seconds, so that it can cover most of the case, while still delay bogus user ingestion.